deploy/mainnet/672_deploy_arb_x_chain_gauges.ts

```diff
@@ -0,0 +1,54 @@
+import { DeployFunction } from "hardhat-deploy/types"
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { convertGaugeNameToSalt, getWithName } from "../../test/testUtils"
+import { CHAIN_ID } from "../../utils/network"
+
+/*
+ * Deploy Root & Child Gauges for Arbitrum using RootGaugeFactory.
+ *
+ * This script calls `RootGaugeFactory.deploy_child_gauges()` for each LPToken on Arbitrum chain ID.
+ * This will initiate a deploy on Arbitrum side using AnyCallV6 which will then
+ * call `RootGaugeFactory.deploy_gauge()` on Ethereum Mainnnet Chain.
+ *
+ * The entire process is expected to take at least 20 minutes after initial txs are
+ * confirmed on Ethereum Mainnet.
+ *
+ * Progress can be tracked by anyswap explorer: https://anyswap.net/
+ */
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, getNamedAccounts, ethers } = hre
+  const { get, execute, deploy, log } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  const executeOptions = {
+    log: true,
+    from: deployer,
+  }
+
+  const lpTokenNameToRegistryName: Record<string, string> = {
+    SaddleArbUSDPoolLPToken: "ArbUSD",
+    SaddleArbUSDPoolV2LPToken: "ArbUSDV2",
+    SaddleArbUSDSMetaPoolLPToken: "USDS-ArbUSDV2",
+    SaddleFRAXBPPoolLPToken: "FRAX-USDC-BP",
+    SaddleFRAXUSDTMetaPoolLPToken: "FRAXBP-USDT",
+    SaddleFRAXUSDsMetaPoolLPToken: "FRAXBP-USDs",
+    SaddleUSXFRAXBPMetaPoolLPToken: "USX-FRAXBP",
+  }
+
+  // For each LP token, call deploy_child_gauge
+  for (const lpTokenName in lpTokenNameToRegistryName) {
+    const lpToken = await getWithName(lpTokenName, "arbitrum_mainnet")
+    const lpTokenRegistryName = lpTokenNameToRegistryName[lpTokenName]
+    await execute(
+      "RootGaugeFactory",
+      executeOptions,
+      "deploy_child_gauge(uint256,address,bytes32,string)",
+      CHAIN_ID.ARBITRUM_MAINNET, // chainId
+      lpToken.address, // lpToken address on Arbitrum
+      convertGaugeNameToSalt(lpTokenRegistryName), // salt
```

Comment:
Might run into conflicts on RootGauge deployments, when just using `lpTokenRegistryName` as salt, since we use the same pool names in registries on multiple chains (eg. `FRAXBP-USDT` is used on both on Arb and Opt). Maybe include destination chain name in salt to make it unique

Reply:
https://github.com/saddle-finance/saddle-contract/blob/241e2fc3f440b0cd69e678a0b46607ee31d98df9/contracts/xchainGauges/ChildGaugeFactory.vy#L161-L163

https://github.com/saddle-finance/saddle-contract/blob/241e2fc3f440b0cd69e678a0b46607ee31d98df9/contracts/xchainGauges/RootGaugeFactory.vy#L99-L103

Luckily the addresses are generated with a hash w/ the chainID of the child gauge. So as long as there are not duplicate names on the **same chainID** we should be good.